### ISSUE TITLE
Compatible with previous PyTorch version

### ISSUE THREAD

**Issue by Yancey1989** [llvm/torch-mlir#1009]
Hi torch-mlir team,

I came from the [BladeDISC](https://github.com/alibaba/BladeDISC) repo.  We noticed that torch-mlir always follows the latest PyTorch version, and I want to know does torch-mlir team has any plan or any suggestion to compilable with the previous PyTorch version? As the following example, some Aten symbols are different between different PyTorch version:
-  [gelu(Tensor self, *, str approximate='none') -> Tensor in PyTorch 1.12](https://github.com/pytorch/pytorch/blob/v1.12.0-rc6/aten/src/ATen/native/native_functions.yaml#L3989) 
- [gelu(Tensor self) -> Tensor in PyTorch 1.8.1](https://github.com/pytorch/pytorch/blob/v1.8.1/aten/src/ATen/native/native_functions.yaml#L3273 )

The TorchDialect only described the latest version, ref: https://github.com/llvm/torch-mlir/blob/main/include/torch-mlir/Dialect/Torch/IR/GeneratedTorchOps.td#L2576

**Comment by powderluv** [comment]
hi @Yancey1989 we are actively working on getting our release process ready for targeting stable and nightly PyTorch -- along with snapshots that downstream users can branch and build on.  However it is unlikely we will support older versions of PyTorch like 1.8.1 etc Are you using 1.8.1 as an example or do you really care about 1.8.1 specifically ?

@rdadolf @silvasean

**Comment by Yancey1989** [comment]
Yes, we are care about the older versions, because our customers do not always deploy the latest  PyTorch version in the production environment, BladeDISC needs to be compatible with the previous PyTorch, e.g. 1.7, 1.8.1, 1.10, etc.

**Comment by silvasean** [comment]
@Yancey1989 -- thanks for raising this.

If you are using the JIT IR importer (and not LTC), I have had good success with serializing the JIT IR module (TorchScript module) to disk with the older PyTorch version, and then reloading it with the new PyTorch version. I think they have some autoupgrade mechanism. This would require a lot of care with managing shared libraries and Python packages, but I think it could be made to work. And of course it is not great to have to serialize the module.

Are you currently blocked on this?

@rdadolf -- You said it better than I could. I think you captured all the essential difficulties here.

**Comment by Yancey1989** [comment]
@rdadolf @silvasean thanks for your reply.

@silvasean , I think your advice is very useful. And we are actually blocked on this issue to switch mho conversion to torch-mlir world.  We will do some experiments and keep updating the process on this issue, thanks!
